src/new.rs

```diff
@@ -613,3 +630,21 @@ fn validate_name(name: &str) -> Result<(), Error> {
         Ok(())
     }
 }
+
+fn get_foldername(path: &str) -> Option<String> {
```

Comment:
```suggestion
fn get_foldername(path: &str) -> Result<String, Error> {
```

I think we want this function to return a result￼￼￼. This way if the function is able to determine what the name should be then we have just a string which we can safely use. If it is unable to determine what the name is that we should use it returns in error and the error can be presented to the user to say they have supplied incorrect details￼.

Reply:
got something like this working, do you think it'd be ok?

```rust
fn get_foldername(path: &str) -> Result<String, Error> {
    match path {
        "." => env::current_dir()
            .gleam_expect("invalid folder")
            .file_name()
            .and_then(|x| x.to_str())
            .map(ToString::to_string)
            .ok_or(Error::UnableToFindProjectRoot{path: path.to_string()}),
        _ => Path::new(path)
            .file_name()
            .and_then(|x| x.to_str())
            .map(ToString::to_string)
            .ok_or(Error::UnableToFindProjectRoot{path: path.to_string()}),
    }
}
```